src/constraint/CollisionFree.cpp

```diff
@@ -42,16 +43,39 @@ bool CollisionFree::isSatisfied(
         groups.second.get(),
         mCollisionOptions,
         &collisionResult);
+
     if (collision)
+    {
+      if (_outcome != nullptr)
+      {
+        auto collisionOutcome = static_cast<CollisionFreeOutcome*>(_outcome);
+        auto collidingBodies = collisionResult.getCollidingBodyNodes();
+        for (const auto& elem : collidingBodies)
```

Comment:
Maybe this should iterate over `CollisionResult::getContacts()` instead. That will allow us to specify which _pairs_ of bodies are in collision, rather than squashing that information into a list of bodies that are in collision.

This isn't a problem now because our default `CollisionOption` only detects a maximum of one contact between two bodies, but we may as well support multiple contacts on the off-chance that becomes helpful.